src/Model/IndexInfo.php

```diff
@@ -110,6 +200,29 @@ public function getVersion()
         return (integer) $this->info['v'];
     }
 
+    /**
+     * Return the weights document for text indexes.
+     *
+     * @return document
+     */
+    public function getWeights()
+    {
+        if (empty($this->info['weights'])) {
+            return self::DEFAULT_WEIGHT;
+        }
+        return $this->info['weights'];
+    }
+
+    /**
+     * Return whether background is true.
+     *
+     * @return boolean
+     */
+    public function isBackground()
+    {
+        return ! empty($this->info['background']);
```

Comment:
As you can set the ``background`` option to ``false``, this line is not right, as it would return ``true`` even if ``background`` is set to false, as in:

```
> db.foo1.createIndex( { test2: 'text' }, { background: false } );
{
	"createdCollectionAutomatically" : true,
	"numIndexesBefore" : 1,
	"numIndexesAfter" : 2,
	"ok" : 1
}
> db.foo1.getIndexes();
[
…
	{
		"v" : 2,
		"key" : {
			"_fts" : "text",
			"_ftsx" : 1
		},
		"name" : "test2_text",
		"ns" : "test.foo1",
		"background" : false,
		"weights" : {
			"test2" : 1
		},
		"default_language" : "english",
		"language_override" : "language",
		"textIndexVersion" : 3
	}
]
```

Reply:
Teaching me PHP now huh ;-) You're absolutely right. My mistake.